pkg/credentialprovider/aws/aws_credentials_test.go

```diff
@@ -52,112 +69,221 @@ func (p *testTokenGetter) GetAuthorizationToken(input *ecr.GetAuthorizationToken
 	output := &ecr.GetAuthorizationTokenOutput{
 		AuthorizationData: []*ecr.AuthorizationData{data},
 	}
-
 	return output, nil //p.svc.GetAuthorizationToken(input)
 }
 
-func TestEcrProvide(t *testing.T) {
+func TestParseRepoURLPass(t *testing.T) {
```

Comment:
this is a good candidate for a table test to exercise the options in the regex (and demonstrate we're not vulnerable to suffix spoofing)

* registry id too short/long/invalid characters
* with/without fips
* region too short/long/invalid chars
* with/without china suffix
* with/without additional invalid host suffixes (e.g. `...amazonaws.com.hacker.com`)
* with/without port number?